src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs

```diff
@@ -2658,7 +2658,7 @@ public static Tensor<T> PermuteDimensions<T>(this Tensor<T> tensor, params ReadO
 
                 if (dimensions.IsEmpty)
                 {
-                    lengths = tensor._lengths.Reverse().ToArray();
+                    lengths = Enumerable.Reverse(tensor._lengths).ToArray();
```

Comment:
@michaelgsharp, subsequently we should avoid using LINQ in these cases.

Reply:
This instance of LINQ usage was addressed in the changes from https://github.com/dotnet/runtime/pull/109405, as it was hindering the source-build codeflow. Perhaps we should consider removing LINQ usage entirely from tensor computation (to ensure more efficient and optimized code).

Reply:
https://github.com/dotnet/runtime/pull/109415